pkg/kubectl/cmd/util/helpers.go

```diff
@@ -122,6 +122,11 @@ func checkErrWithPrefix(prefix string, err error) {
 // checkErr formats a given error as a string and calls the passed handleErr
 // func with that string and an kubectl exit code.
 func checkErr(prefix string, err error, handleErr func(string, int)) {
+	// unwrap aggregates of 1
+	if agg, ok := err.(utilerrors.Aggregate); ok && len(agg.Errors()) == 1 {
```

Comment:
I've stumbled upon this when cherry-picking ScheduledJobs, how about moving this to `pkg/util/errors/errors.go#NewAgregate` so that we could unwrap aggregates upon creating. Eg. [here](https://github.com/kubernetes/kubernetes/blob/06a23dddbee223a6d3a0f9b3eff4bc15b7b17277/pkg/kubectl/cmd/get.go#L364) `r.Infos()` returns `errors.aggregate` which we add to an array of errors and then create a new aggregate. How about the `errors.NewAggregate` would actually unpack the error if it's just one error?